RuriLib/Blocks/Interop/Methods.cs

```diff
@@ -40,14 +41,57 @@ public static string ShellCommand(BotData data, string executable, string argume
          * of writing C# code that calls these methods where necessary once it's transpiled.
          */
 
-        public static async Task<T> InvokeNode<T>(BotData data, string scriptFile, object[] parameters)
+        public static async Task<T> InvokeNode<T>(BotData data, (string, string) scriptOrFile, object[] parameters, bool isScript = false)
         {
             data.Logger.LogHeader();
-            var result = await StaticNodeJSService.InvokeFromFileAsync<T>(scriptFile, null, parameters, data.CancellationToken).ConfigureAwait(false);
+            T result;
+
+            if (isScript)
+            {
+                result = await InvokeFromStringOrCache<T>(data, scriptOrFile, parameters);
+            }
+            else
+            {
+               
+                result = await InvokeFromFile<T>(data, scriptOrFile.Item2 /*hash*/, parameters);
+            }
+
             data.Logger.Log($"Executed NodeJS script with result: {result}", LogColors.PaleChestnut);
             return result;
         }
+        private static async Task<T> InvokeFromStringOrCache<T>(BotData data, (string, string) script, object[] parameters)
```

Comment:
Please separate methods with a blank line, according to C# coding style guidelines. Also, accept two individual parameters for `script` and `scriptHash` instead of a tuple.